### DISCUSSION TITLE
Any way to detect domain ?

### DISCUSSION THREAD

**Discussion by digitalnature** [QuorumDMS/ftp-srv#114] [Q&A]
If someone logs in using the domain name, like `mysite.com:21`, is there a way to find that out within the server ?

I noticed there's a `domain` property in the data object passed in the login event, but it's always null.

**Comment by dralletje** [comment]
Sadly, that is not sent to FTP server, so there is no way to find it out :(
It would be cool if it were there though, so we could use it to log them in to the right website

**Comment by digitalnature** [comment]
yeah i think ftp over tls is also not possible if the server is hosting multiple sites..

**Comment by trs** [comment]
Looks like you've resolved this?

**Comment by digitalnature** [comment]
no. I thought it was not part of the FTP protocol, but it looks like there's a HOST command in the standard

https://tools.ietf.org/html/rfc7151

Could you add support for it in ftp-srv ?

**Comment by lezsakdomi** [comment]
I've created a draft PR (#169)  for RFC 7151. Some meta things like tests and CLI are missing yet and it is not fully RFC compliant as it does not check for valid hostname syntax and also does not check whether the IP address is ours.

I think it the PR could be merged even without that full compilance thing, I plan to implement them later when I have more time.
Could someone take a look at that to add proper tests, plan the CLI and point out my blatant mistakes? (or maybe even fix my two bugs already discovered)

**Comment by lezsakdomi** [comment]
By the way, TLS already exchanges domain name, so this `HOST` command is unneded in the context of this particular issue.

**Comment by Shinrai** [comment]
TLS does handle domain. However without being able to pass more than the TLS context object you aren’t able to set the correct TLS context object for each domain.
